tests/ui/iter_overeager_cloned.stderr

```diff
@@ -130,5 +130,21 @@ LL |             iter.cloned().filter(move |S { a, b }| **a == 1 && b == &target
    |                 |
    |                 help: try: `.filter(move |&S { a, b }| **a == 1 && b == &target).cloned()`
 
-error: aborting due to 15 previous errors
+error: unneeded cloning of iterator items
+  --> $DIR/iter_overeager_cloned.rs:60:13
+   |
+LL |     let _ = vec.iter().cloned().map(|x| x.len());
+   |             ^^^^^^^^^^--------------------------
+   |                       |
+   |                       help: try: `.map(|x| x.len())`
+
+error: unneeded cloning of iterator items
+  --> $DIR/iter_overeager_cloned.rs:65:13
+   |
+LL |     let _ = vec.iter().cloned().for_each(|x| assert!(!x.is_empty()));
+   |             ^^^^^^^^^^----------------------------------------------
```

Comment:
Maybe we need to fix the origin .stderr first

Reply:
Yep, just checked, it is a lint-scoped problem. It's problematic if the user isn't using a tool like `rustfix` to apply the suggestions for them. Could you fix it or do you think it is out of scope for this PR?

Reply:
A quick review, some other files suffer same problems:

tests/ui/float_equality_without_abs.stderr
tests/ui/unnecessary_lazy_eval_unfixable.stderr
tests/ui/unnecessary_lazy_eval.stderr

Reply:
After some reflexing on this discussion, I just realized this is total bikeshedding. It isn't worth pausing a lint just because of a stylistic error. I'll discuss this stylistic choice in the Zulip chat on the next meeting and decide from that. This discussion can be resolved.